imgreg/multi_spect_reg_config.py

```diff
@@ -1,167 +1,260 @@
+                config_dict[p_map]["opt_scale_sigmas"], float
+            )
+            self.param_map[ch_name]["opt_final_metric_min"] = float(
+                config_dict[p_map]["opt_final_metric_min"]
+            )
+
+            self.param_map[ch_name]["evol_epsilon"] = float(
+                config_dict[p_map]["evol_epsilon"]
+            )
+            self.param_map[ch_name]["evol_iterations"] = int(
+                config_dict[p_map]["evol_iterations"]
+            )
+            self.param_map[ch_name]["evol_init_rad"] = float(
+                config_dict[p_map]["evol_init_rad"]
+            )
+            self.param_map[ch_name]["evol_growth_fact"] = float(
+                config_dict[p_map]["evol_growth_fact"]
+            )
+            self.param_map[ch_name]["evol_shrink_fact"] = float(
+                config_dict[p_map]["evol_shrink_fact"]
+            )
+
+        # Load Per-Channel Camera Settings
+        self.camera_params = {}
+        for ch_name in self.ordered_channel_names:
+            self.load_cam_config(config_dict, ch_name)
+
+        if "OPTIONS" in config_dict.keys():
+            if "remove_partial_edges" in config_dict["OPTIONS"].keys():
+                self.remove_partial_edges = (
+                    config_dict["OPTIONS"]["remove_partial_edges"] == "True"
+                )
+            else:
+                self.remove_partial_edges = False
+
+            if "rgb_6x" in config_dict["OPTIONS"].keys():
+                self.rgb_6x = self.clean_string(config_dict["OPTIONS"]["rgb_6x"])
+                # check that the rgb_6x channel name is also in the ordered channel names
+                assert self.rgb_6x in self.ordered_channel_names
+            else:
+                self.rgb_6x = None
+        else:
+            self.remove_partial_edges = False
+            self.rgb_6x = None
+
+    def load_image_dict(self, data_set_paths_dict):
+        """Load images that appear in all channels."""
+        self.img_path_dict = {}
+        for ch_name in self.ordered_channel_names:
+            file_list = os.listdir(data_set_paths_dict[ch_name])
+            # print("Ch %s found %i files"%(ch_name, len(file_list)))
+            # identify image type
+            if file_list[0].endswith(".jpg"):
+                if ch_name == self.fixed_channel_name:
+                    self.image_extension = ".jpg"
+            elif file_list[0].endswith(".tif") or file_list[0].endswith(".tiff"):
+                if ch_name == self.fixed_channel_name:
+                    self.image_extension = ".tif"
+            else:
+                print(
+                    f"Image file {file_list[0]} is not of a supported type. (.jpg, ,tif, .tiff)"
+                )
+                raise TypeError
+            for file_name in file_list:
+                img_str_list = list(file_name.replace(".", "_").split("_"))
+                for img_str in img_str_list:
+                    if img_str.isdigit:
```

Comment:
This should be a method, right? `img_str.isdigit()`

Reply:
I forgot to push this! Good catch.